openmdao/core/driver.py

```diff
@@ -468,6 +490,17 @@ def _get_voi_val(self, name, meta, remote_vois, driver_scaling=True, rank=None):
                     val = np.empty(size)
 
                 comm.Bcast(val, root=owner)
+
+        elif distributed:
+            local_val = model._get_val(name, flat=True)
+            if indices is not None:
+                val = val[indices]
```

Comment:
This val gets overwritten below by `val = np.zeros(np.sum(sizes))`, so indices are not being handled.

Reply:
Should be local_val, but this is an obvious coverage hole, so i need to add a test.

Reply:
Looks like defining indices on a distributed response doesn't work yet. I think it makes sense for that to be a separate story, and might require some api discussiion.

Reply:
I think you'll need to treat indices as global in the same way as we treat src_indices, which means you'll have to check what part of the indices refer to a given proc and only gather those from each proc.